### ISSUE TITLE
Enabling `write_id3`  prevents song edits from taking effect.

### ISSUE THREAD

**Issue by wagnered** [ampache/ampache#2468]
`write_id3` calls catalog::update_song_from_tags(). The instance of song it creates seems to have the old data thus overwriting the changes made earlier in Song::update().

**Comment by wagnered** [comment]
0k, I got it.

`Song::write_id3()`  calls  `Catalog::update_media_from_tags($this)` but `$this` hasn't been updated with the new values
 Also because the database has already been updated, calling the catalog function  seems redundant.

Actually because `$this` hasn't been updated,  the new metadata isn't being written to the file anyway.

**Comment by wagnered** [comment]
Question: if a song gets assigned to a new artist or an existing artist, shouldn't either of their names be updated in the file?   The same goes for album, album artist, etc?

**Comment by lachlan-00** [comment]
definitely, any change on an object that's used in multiple tables like artist/album should get updated.

But i think because the name values are in their own table and ID is used for joins it should only need one update

**Comment by lachlan-00** [comment]
updating the file tags i would find exact matches and replace them. in those sort of fields

**Comment by wagnered** [comment]
What I'm doing is just extending what exists. I added 'disk' to $song->write_id3() and the ability of $album-update() to update any of it's metadata without creating a new id.  The song object does that when necessary.

I just finished adding the capability of writing an artist image to the the tags. Before, only covers(front) could be written.  The code will currently gather all images from the tags, but not sure yet what Ampache does with them.

It's super slow,. I'd hate to see how long it will take with  a large number of songs.  I will see what it will take to disable `write_id3` on catalog creation.

> But i think because the name values are in their own table and ID is used for joins it should only need one update

> updating the file tags i would find exact matches and replace them. in those sort of fields

Not sure what all that means or how it applies to writing metadata to the file tags (Or maybe I do).  Ampache is configured to overwrite existing tags, so the getID3 library creates a temp file, lays out the tags then appends the audio part of the file to the new tag leader.  So all the updated song's tags must be submitted to the tag writer at the same time.
